src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs

```diff
@@ -1776,6 +1776,12 @@ public void WriteMetadataAndIL(PdbWriter nativePdbWriterOpt, Stream metadataStre
                 {
                     EmbedSourceLink(module.SourceLinkStreamOpt);
                 }
+
+                if (_deterministic)
```

Comment:
>_deterministic [](start = 20, length = 14)

I'd emit it regardless of deterministic switch value. This info is useful even if the compilation is not fully deterministic.